aopy/data/bmi3d.py

```diff
@@ -564,4 +566,127 @@ def get_source_files(preproc_dir, subject, te_id, date):
             |** data_dir (str):** directory where the source files were located
     '''
     exp_data, exp_metadata = load_preproc_exp_data(preproc_dir, subject, te_id, date)
-    return exp_metadata['source_files'], exp_metadata['source_dir']
\ No newline at end of file
+    return exp_metadata['source_files'], exp_metadata['source_dir']
+
+def concat_trials(preproc_dir, subjects, ids, dates, trial_start_codes, trial_end_codes, target_codes, 
```

Comment:
I think either the title should specify that this is for task data or the function should be written to concatenate trials of any variable name in the preprocessed data (including neural data such as lfp_data).

Reply:
I did try to include lfp segments but it wasn’t very easy and probably doesn’t make much sense. I think this function should be limited to kinematics for now. It might be useful to separate out the trial concatenation from the kinematics part, so we could reuse the trial concatenation for other kinds of data. But maybe outside the scope of this pull request.

Reply:
I think it is fine to keep the pull request as is, but I think the function name should be something like "concat_kin_trials" or "concat_trials_kindata" so the major restriction is obvious.

Reply:
I see now that this is addressed at in the comments below.